content/blog/integrating-python-3-with-fips-enabled-openssl-3-1.md

```diff
@@ -0,0 +1,149 @@
+> copy `C:\SSLout\DLL\x64\Release\bin\*.*` to `openssl-bin-3.1.0\amd64`\
+> copy `C:\SSLout\DLL\x64\Release\include` directory to `openssl-bin-3.1.0\amd64`\
+> copy `C:\SSLout\DLL\x64\Release\include\openssl\applink.c` to `openssl-bin-3.1.0\amd64\include`\
+> copy `C:\SSLout\DLL\x64\Release\lib\ossl-modules\*.*` to `openssl-bin-3.1.0\amd64`
+
+Step 9 Modify `PCbuild/openssl.props` as shown below:
+
+![](/img/openssl_settings.jpg)
+
+Step 10 Open `PCbuild/python.props` and change the entries as shown below:
+
+![](/img/openssl_settings2.jpg)
+
+Step 11 Open Python Solution under `PCbuild/pcbuild.sln` in VS 2015.
+Change the Link settings of `_hashlib` and `_ssl` projects under Python as shown below:
+
+![](/img/openssl_vs_settings.jpg)
+
+Step 12 Now, build `_hashlib.pyd` and `_ssl.pyd` in VS 2015.
+Step 13 Copy these built .pyd files from `\PCbuild\amd64\` to a Python binary installation directory `C:\python311\DLLs`.
+Step 14 Start Python and use these commands to check the OpenSSL 3.1 version.
+
+![](/img/openssl-after.jpg)
+
+Step 15 Now, to run Python with FIPS enabled OpenSSL, create `openssl.cnf` and `fipsmodule.cnf` file using the below content in `C:\Python311` directory. By using this config file, FIPS will be enabled by default.
+
+> #### openssl.cnf:
+>
+> config_diagnostics = 1\
+> openssl_conf = openssl_init\
+> .include .\fipsmodule.cnf\
+> \[openssl_init]
+> providers = provider_sect\
+> alg_section = algorithm_sect\
+> \[provider_sect]\
+> fips = fips_sect\
+> legacy = legacy_sect\
+> base = base_sect\
+> default = default_sect\
+> \[base_sect]\
+> activate = 1\
+> \[legacy_sect]\
+> activate = 1\
+> \[default_sect]\
+> activate = 1\
+> \[algorithm_sect]\
+> default_properties = fips=yes
+>
+> #### fipsmodule.cnf:
+>
+> \[fips_sect]\
+> activate = 1\
+> conditional-errors = 1\
+> security-checks = 1\
+> module-mac = D4:64:00:E3:CE:34:EE:CE:58:32:12:08:21:6D:64:FD:E3:A6:D4:F0:E6:38:3D:2C:0C:40:1B:50:C8:8F:39:A3
+
+Step 16 Now open command windows as Administrator and execute 
+
+> `set OPENSSL_CONF=C:\Python311\openssl.cnf` 
+
+Step 17 To verify python is enabled with FIPS, both client and server needs to be FIPS mode. Client Windows OS need to be enabled with FIPS with these 2 steps.
+
+> 1. Open `gpedit.msc` on run menu and navigate to `Computer Configuration\Windows Settings\Security Settings\Local Policies\Security Options` and enable the `System cryptography: Use FIPS compliant algorithms for encryption, hashing, and signing` setting.   
+> 2. Open `regedit` on run menu and go to `HKLM\System\CurrentControlSet\Control\Lsa\FipsAlgorithmPolicy\Enabled` and set Enabled to 1.
+
+Step 18 To verify python is enabled with FIPS, run the following commands. 
+
+![](/img/openssl_fips_algo.jpg)
+
+Note that the list crypto algorithms available are more than the crypto algorithms guaranteed.  But all algorithms can be used if Server where Client SSL is connecting is also configured in FIPS mode. 
```

Comment:
add the word "of" prior to "crypto algorithms".
Consider rephrasing the second sentence to:
But all algorithms can be used if the Server where the Client SSL is connecting to is also configured in FIPS mode."

(it's currently missing some articles and prepositions which help the reader understand where you're going from and to)